pkg/api/config_test.go

```diff
@@ -0,0 +1,108 @@
+package api_test
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+)
+
+var searchPattern = regexp.MustCompile(`(\d+)`)
+var replacePattern = []byte(` $1 `)
+
+// padNumbers inserts spaces around numbers within a string
+func padNumbers(s string) string {
+	b := []byte(s)
+	b = searchPattern.ReplaceAll(b, replacePattern)
+	return string(b)
+}
+
+func TestPadNumbers(t *testing.T) {
+	testcases := []struct {
+		input    string
+		expected string
+	}{
+		{input: "single123number", expected: "single 123 number",},
+		{input: "single123numbers345spread", expected: "single 123 numbers 345 spread",},
+		{input: "number_at_end234", expected: "number_at_end 234 ",},
+		{input: "123number_at_start", expected: " 123 number_at_start",},
+		{input: "123number_at_start", expected: " 123 number_at_start",},
+	}
+
+	for _, test := range testcases {
+		got := padNumbers(test.input)
+		if got != test.expected {
+			t.Errorf("padNumbers(%v) != %v, expected %v", test.input, got, test.expected)
+		}
+	}
+}
+
+// Converts a string to lower camel case
+func lowerCamelCase(s string) string {
+	if s == "" {
+		return s
+	}
+
+	prep := s
+	start, rest := s[0], s[1:]
+	if start >= 'A' && start <= 'Z' {
+		prep = strings.ToLower(string(start)) + rest
+	}
+	//  pad all numbers within the string with spaces
+	prep = padNumbers(prep)
+
+	// trim spaces at the edges of the string
+	prep = strings.Trim(prep, " ")
```

Comment:
shouldn't this be a bit up on line 47? to catch the case " ABook"

Reply:
good catch! thanks. added more testcases for this condition.